xds/internal/balancer/clusterresolver/e2e_test/balancer_test.go

```diff
@@ -414,3 +494,946 @@ func (s) TestEDSResourceRemoved(t *testing.T) {
+				return nil
+			}
+			if len(req.GetResourceNames()) == 0 {
+				// This is the case for ACKs. Do nothing here.
+				return nil
+			}
+			select {
+			case edsResourceNameCh <- req.GetResourceNames():
+			case <-ctx.Done():
+			}
+			return nil
+		},
+		AllowResourceSubset: true,
+	})
+	defer cleanup()
+
+	// Start two test backends and extract their host and port. The first
+	// backend belongs to EDS cluster "cluster-1", while the second backend
+	// belongs to EDS cluster "cluster-2".
+	servers, cleanup2 := startTestServiceBackends(t, 2)
+	defer cleanup2()
+	addrs, ports := backendAddressesAndPorts(t, servers)
+
+	// Configure an aggregate cluster, two EDS clusters and only one endpoints
+	// resource (corresponding to the first EDS cluster) in the management
+	// server.
+	const clusterName1 = clusterName + "-cluster-1"
+	const clusterName2 = clusterName + "-cluster-2"
+	resources := e2e.UpdateOptions{
+		NodeID: nodeID,
+		Clusters: []*v3clusterpb.Cluster{
+			makeAggregateClusterResource(clusterName, []string{clusterName1, clusterName2}),
+			e2e.DefaultCluster(clusterName1, "", e2e.SecurityLevelNone),
+			e2e.DefaultCluster(clusterName2, "", e2e.SecurityLevelNone),
+		},
+		Endpoints:      []*v3endpointpb.ClusterLoadAssignment{e2e.DefaultEndpoint(clusterName1, "localhost", []uint32{uint32(ports[0])})},
+		SkipValidation: true,
+	}
+	if err := managementServer.Update(ctx, resources); err != nil {
+		t.Fatal(err)
+	}
+
+	// Create an xDS xdsClient for use by the cluster_resolver LB policy.
+	xdsClient, close, err := xdsclient.NewWithBootstrapContentsForTesting(bootstrapContents)
+	if err != nil {
+		t.Fatalf("Failed to create xDS client: %v", err)
+	}
+	defer close()
+
+	// Create a manual resolver and push a service config specifying the use of
+	// the cds LB policy as the top-level LB policy, and a corresponding config
+	// with a single cluster.
+	r := manual.NewBuilderWithScheme("whatever")
+	jsonSC := fmt.Sprintf(`{
+			"loadBalancingConfig":[{
+				"cds_experimental":{
+					"cluster": "%s"
+				}
+			}]
+		}`, clusterName)
+	scpr := internal.ParseServiceConfig.(func(string) *serviceconfig.ParseResult)(jsonSC)
+	r.InitialState(xdsclient.SetClient(resolver.State{ServiceConfig: scpr}, xdsClient))
+
+	// Create a ClientConn.
+	cc, err := grpc.Dial(r.Scheme()+":///test.service", grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(r))
+	if err != nil {
+		t.Fatalf("Failed to dial local test server: %v", err)
+	}
+	defer cc.Close()
+
+	// Wait for both EDS resources to be requested.
+	func() {
+		for ; ctx.Err() == nil; <-time.After(defaultTestShortTimeout) {
+			select {
+			case n := <-edsResourceNameCh:
+				// A data race is reported between here and the OnStreamRequest
+				// if this code attempts to sort the names without a copy.
+				names := make([]string, len(n))
+				copy(names, n)
+				sort.Strings(names)
+				t.Logf("easwars: sorted names: %v", names)
```

Comment:
I use all caps to find my logs easy to find. Or ctrl f with the log string. Please delete :P